### PR TITLE
test(useScrollLock): improve coverage

### PR DESCRIPTION
**Pull request by mdodgelooker**
Improves coverage mostly with the help of `jest.spyOn`. The changes I made were in `useScrollLock.test` but the file whose coverage I was aiming to improve is `ScrollLock/utils.ts`.

### CODE REVIEW

File: packages/components/src/utils/useScrollLock.test.tsx [hunk 1]
```diff
@@ -82,12 +92,36 @@ describe('useScrollLock', () => {
...
       fireEvent.click(toggle)
       expect(document.body).toHaveStyle({ overflow: 'scroll' })
       document.body.style.overflow = ''
     })
+
+    test('no scrollbar detected', () => {
+      const widthSpy = jest
+        .spyOn(document.documentElement, 'clientWidth', 'get')
+        // Make it bigger than jest default for innerWidth: 1024
+        .mockImplementation(() => 1025)
```
Comment (mdodgelooker) [inline packages/components/src/utils/useScrollLock.test.tsx:113]: The scrollbar offset detection compares `innerWidth` to `clientWidth` and when the latter is bigger, that means there _is_ a scrollbar offset.

File: packages/components/src/utils/useScrollLock.test.tsx [hunk 2]
```diff
@@ -100,4 +134,27 @@ describe('useScrollLock', () => {
       ]
     `)
   })
+
+  test('stop scroll', () => {
```
Comment (mdodgelooker) [inline packages/components/src/utils/useScrollLock.test.tsx:138]: Test the stop scroll behavior for other scrollable elements.

File: packages/components/src/utils/useScrollLock.test.tsx [hunk 3]
```diff
@@ -31,16 +31,19 @@ import { useScrollLock, useToggle } from './'
 
 const globalConsole = global.console
 const warnMock = jest.fn()
+const paddingSpy = jest.spyOn(document.body.style, 'paddingRight', 'set')
```
[the remark this answers is not in the document]
Comment (dbchristopher) [reply packages/components/src/utils/useScrollLock.test.tsx:34]: clever!

**Review by dbchristopher — APPROVED** [review]
Wow this looks like a really tricky one to test! Nice work 👍

**Pull request merged**